cfy_manager/components/prometheus/prometheus.py

```diff
@@ -0,0 +1,357 @@
+            POSTGRES_EXPORTER_VERSION))
+    _unpack_archive(archive_file_name, working_dir)
+    _copy_postgres_exporter(working_dir)
+    common.remove(working_dir)
+    logger.notice('PostgreSQL Exporter successfully installed')
+
+
+def _copy_postgres_exporter(src_dir):
+    logger.notice('Copying PostgreSQL Exporter binaries')
+    common.copy(join(src_dir, 'postgres_exporter'), BIN_DIR)
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP,
+                 join(BIN_DIR, 'postgres_exporter'))
+
+
+def _install_rabbitmq_exporter():
+    # As of RabbitMQ 3.8.0 https://github.com/rabbitmq/rabbitmq-prometheus
+    # is also available
+    logger.notice('Installing RabbitMQ Exporter...')
+    working_dir = _create_directory(join(sep, 'tmp', 'rabbitmq_exporter'))
+    archive_file_name = _download_release(working_dir,
+                                          'https://github.com/kbudde',
+                                          RABBITMQ_EXPORTER,
+                                          RABBITMQ_EXPORTER_VERSION)
+    _unpack_archive(archive_file_name, working_dir)
+    _copy_rabbitmq_exporter(working_dir)
+    common.remove(working_dir)
+    logger.notice('RabbitMQ Exporter successfully installed')
+
+
+def _copy_rabbitmq_exporter(src_dir):
+    logger.notice('Copying RabbitMQ Exporter binaries')
+    common.copy(join(src_dir, 'rabbitmq_exporter'), BIN_DIR)
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP,
+                 join(BIN_DIR, 'rabbitmq_exporter'))
+
+
+def _deploy_configuration():
+    logger.info('Initializing Prometheus...')
+    _deploy_services_configuration()
+    _copy_prometheus_configuration()
+    _copy_prometheus_rules()
+    service.reload(PROMETHEUS,
+                   append_prefix=False, ignore_failure=True)
+    service.reload(POSTGRES_EXPORTER,
+                   append_prefix=False, ignore_failure=True)
+    service.reload(RABBITMQ_EXPORTER,
+                   append_prefix=False, ignore_failure=True)
+
+
+def _deploy_services_configuration():
+    logger.notice('Adding Prometheus service configuration...')
+    # TODO: use files.deploy instead of common.copy here:
+    files.deploy(join(CONFIG_PATH, 'prometheus.service'),
+                 join(SYSTEMD_CONFIG_DIR, 'prometheus.service'))
+    files.deploy(join(CONFIG_PATH, '{0}.service'.format(NODE_EXPORTER)),
+                 join(SYSTEMD_CONFIG_DIR,
+                      '{0}.service'.format(NODE_EXPORTER)))
+    files.deploy(join(CONFIG_PATH, '{0}.service'.format(BLACKBOX_EXPORTER)),
+                 join(SYSTEMD_CONFIG_DIR,
+                      '{0}.service'.format(BLACKBOX_EXPORTER)))
+    files.deploy(join(CONFIG_PATH, '{0}.service'.format(POSTGRES_EXPORTER)),
+                 join(SYSTEMD_CONFIG_DIR,
+                      '{0}.service'.format(POSTGRES_EXPORTER)))
+    files.deploy(join(CONFIG_PATH, '{0}.service'.format(RABBITMQ_EXPORTER)),
+                 join(SYSTEMD_CONFIG_DIR,
+                      '{0}.service'.format(RABBITMQ_EXPORTER)))
```

Comment:
@mateumann I think we can reduce this and move them inside a for loop

Reply:
done